Explorer/Assets/DCL/Character/CharacterMotion/Systems/TeleportPositionCalculationSystem.cs

```diff
@@ -0,0 +1,165 @@
+﻿using Arch.Core;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.CharacterCamera;
+using DCL.CharacterMotion.Components;
+using DCL.Diagnostics;
+using DCL.Ipfs;
+using DCL.RealmNavigation;
+using ECS.Abstract;
+using UnityEngine;
+using Utility;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine.Pool;
+using Random = System.Random;
+
+namespace DCL.Character.CharacterMotion.Systems
+{
+    [UpdateInGroup(typeof(PostRenderingSystemGroup))]
+    [LogCategory(ReportCategory.MOTION)]
+    public partial class TeleportPositionCalculationSystem : BaseUnityLoopSystem
+    {
+        private static readonly Random RANDOM = new ();
+
+        private readonly Entity playerEntity;
+
+        private SingleInstanceEntity? cameraCached;
+        private SingleInstanceEntity cameraEntity => cameraCached ??= World.CacheCamera();
+
+        public TeleportPositionCalculationSystem(World world, Entity playerEntity) : base(world)
+        {
+            this.playerEntity = playerEntity;
+        }
+
+        protected override void Update(float t)
+        {
+            ref PlayerTeleportIntent teleportIntent = ref World.TryGetRef<PlayerTeleportIntent>(playerEntity, out bool hasTeleportIntent);
+
+            if (hasTeleportIntent && !World.Has<TeleportPosition>(playerEntity))
```

Comment:
You have to handle the `TeleportPosition` because they are separated from `PlayerTeleportIntent`. As far as I see it, one cannot live without the other; so I would merge them

Reply:
I was considering TeleportPosition as a result of TeleportIntent (so it is became more like a TeleportRequest). Probably I need to move Parcel there and call it  TeleportTarget or TeleportResult

Reply:
embeded in TeleportIntent